packages/headless/src/composables/tooltip/createTooltip.ts

```diff
@@ -1,19 +1,116 @@
+import { computed, ref, unref, watchEffect, type MaybeRef } from "vue";
 import { createId } from "../..";
 import { createBuilder } from "../../utils/builder";
 
-export const createTooltip = createBuilder(() => {
+export type CreateTooltipOptions = {
+  trigger: MaybeRef<TooltipTrigger | boolean>;
+  /**
+   * Number of milliseconds to use as debounce when showing/hiding the tooltip
+   * with openMode "hover".
+   *
+   * @default 0
```

Comment:
I would prefer an API that doesn't allow a delay to be set when open type is not "hover"

E.g.
```ts
type CreateTooltipOptions = {
    open: {
    type: 'hover';
    delay: number;
  } | "hover" | "click" | boolean;
}
```
or some using Generics